File: day01/part02.hs
import Control.Monad
import Data.List

main :: IO ()
main =
  let mul3 (a, b, c) = a * b * c
   in do
        text <- readFile "day01/input.txt"
        let solutions = map mul3 $ findTriples $ map read $ words text
        forM_ solutions print

findTriples :: [Integer] -> [(Integer, Integer, Integer)]
findTriples list =
  [ (a, b, c)
    | (i, a) <- withIndex list,
      (j, b) <- withIndex $ drop i list,
      c <- drop j list,
      a + b + c == 2020
  ]

withIndex :: [b] -> [(Int, b)]
withIndex list =
  let fn a b = (a + 1, (a, b))
   in snd $ mapAccumL fn 0 list

File: day01/part02.js
const fs = require("fs");

const input = fs
  .readFileSync("day01/input.txt", "utf-8")
  .trim()
  .split("\n")
  .map(Number);
for (const [i, n] of input.entries()) {
  for (const [j, m] of input.slice(i + 1).entries()) {
    for (const o of input.slice(j + 1)) {
      if (n + m + o === 2020) {
        console.log(n * m * o);
      }
    }
  }
}
